docs/blog/_posts/2020-06-16-25th-dotty-milestone-release.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: blog-page
+title: Announcing Dotty 0.25.0-RC2 - speed-up of givens
+author: Anatolii Kmetiuk
+authorImg: /images/anatolii.png
+date: 2020-06-16
+---
+
+Hello! We are excited to announce 0.25.0-RC2 of Dotty. In this version, following feedback of the community, we have improved compilation speeds when programming with givens. We have also made progress towards better cross-compilation of macro definitions between Scala 2 and Scala 3.
+
+You can try out this version right now, from the comfort of your SBT, by visiting the [home page](https://dotty.epfl.ch/) and scrolling down to the "Create a Dotty Project" section.
+
+Alternatively, you can try this version of Scala online via [Scastie](https://scastie.scala-lang.org/). Once you're there, click "Build Settings" and set "Target" to "Dotty".
+
+Enjoy the ride🚀!
+
+<!--more-->
+# Progress on macro compatibility with Scala 2
+One of the most challenging parts of the migration from Scala 2 to Scala 3 are macros. Scala 3 has re-implementing the entire metaprogramming mechanism from scratch, and so all the Scala 2 macros must also be rewritten from scratch using the Scala 3 metaprogramming capabilities.
+
+Under these conditions, cross-compilation of a library between Scala 2 and Scala 3 also becomes more challenging since you need to split your sources between the two versions.
+
+To address this challenge, we are currently working on a solution to allow macro definitions of Scala 2 and Scala 3 macros to reside in the same file. This co-habitation will look as follows:
+
+```scala
+object Macro1 {
+  import scala.language.experimental.macros
+  def lineNumber: Int = macro LineNumberMacroScala2.thisLineNumberImpl
+  inline def lineNumber: Int = ${ LineNumberMacroScala3.thisLineNumberExpr }
+}
+```
+
+Currently, this is a work in progress. You can dive deeper into the current state of things by looking at PR [#8811](https://github.com/lampepfl/dotty/pull/8811).
+
+# Speed up the compiler when programming with givens
+Dotty brings to the user an enhanced error reporting when it comes to programming with givens. This better error reporting, however, proved to be a trade-off. Sometimes it takes an unreasonable amount of time and space to compute a quality suggestion to the end user.
+
+A number of users reported considerable slowdowns when it comes to programming with implicits. To address this issue, we have modified the logic for given imports suggestion. We introduced a per-run budget of 10 seconds, so it is guaranteed that it won't take longer than that time to compute all the given imports.
+
+This default budget is configurable via a compiler flag `-Ximport-suggestion-timeout`.
+
+This change should speed-up the compiler when it comes to programming with givens.
```

Comment:
I believe speed-up with a hyphen is a noun; in this context the hyphen should be omitted.